cloudmarker/stores/emailstore.py

```diff
@@ -0,0 +1,68 @@
+"""email store plugin."""
+
+
+import logging
+from email.mime.multipart import MIMEMultipart
+from email.utils import COMMASPACE, formatdate
```

Comment:
`COMMASPACE` & `formatdate` are used only once in the code, and in this framework, we do not import functions by convention. We only import util modules and then we write `<module>.function()`.
So this line can be re-written as
```python
import email
```
And the call can be made like,
```python
message['Date'] = email.utils.formatdate(localtime=True)
```
This will also give us more clarity that `formatdata` from module `email.utils` is used here. The same can applied to `COMMASPACE` as well.

Reply:
At some level it makes sense to follow this approach. But I googled about both the approaches and found out that there are respective pros and cons of using the respective import convention.

But then here you are trying to import full `email` package instead of just import utils. The main question is the use of any particular lib which you are trying to import. It doesn't make sense to import full email package when I am just using COMMASPACE, formatedate from the utils packages. All the modules inside email is getting referenced which are not used at all.

So I believe, we should only import specific lib which we are using.

Reply:
> It doesn't make sense to import full email package when I am just using COMMASPACE, formatedate from the utils packages.

Why does it not make sense? Please provide some justification.

> All the modules inside email is getting referenced which are not used at all.

We have to import `email.utils` (not the `email` package). If we want to use something like `COMMASPACE` or `formatdate`, the Python language requires that we import the `email.utils` module. Importing the `email` package won't make `COMMASPACE` or `formatdate` available.

So if we want to use something like `COMMASPACE` or `formatdate`, there are three popular options:

1. `from email.utils import COMMASPACE, formatdate` followed by using `COMMASPACE` an `formatdate` later in the code.
2. `import email.utils` followed by using `email.utils.COMMASPACE` and `email.utils.formatdate`.
3. `from email import utils` followed by using `utils.COMMASPACE` and `utils.formatdate`

Option 1 violates our coding convention. We avoid option 1. The reason for avoiding option 1 is the benefits offered by options 2 and 3 discussed below.

Option 2 can sometimes be too verbose because we have to write the fully qualified indentifier every time we use it.

Option 3 is preferred usually. When we see something like `utils.formatdate` in the code, it tells us clearly that `formatdate` is not written in the current module but instead imported from another module. This kind of contextual  information is helpful in reducing cognitive load while debugging a complex issue. For example, we can quickly eliminate `formatdate` as a likely suspect for a bug because we know it has come from a well tested standard library.

Also, note that in all three cases, the entire `email.utils` package is imported anyway.

Having said that in this specific case, I would choose Option 2 even if it is a little too verbose. The reason is that `utils` is such a common name that `utils.formatdate` still keeps us confused whether it is our `utils` module or a standard/thirdparty library's `utils` module. On the other hand, `email.utils.formatdate` is very explicit and tells us accurately where `formatdate` is defined.

Reply:
Got the concept of importing functions and classes. So whatever you mentioned now makes sense.